src/actions.js

```diff
@@ -449,4 +456,50 @@ export class DropItemAction extends ItemAction {
     toString() {
         return { action: "DropItemAction", args: { inventorySlot: this.inventorySlot }};
     }
+}
+
+export class EquipAction extends Action {
+    constructor(entity, inventorySlot) {
+        super(entity);
+        this.inventorySlot = inventorySlot;
+    }
+
+    perform(doAction) {
+        if (doAction) {
+            // check if entity has equipment
+            const messageLog = this.getEngine().ui.messageLog;
+            const equippable = this.entityRef.inventory.items[this.inventorySlot];
+            if(this.entityRef.equipment && equippable.equippable) {
+                this.results = this.entityRef.equipment.toggleEquip(equippable);
+                const self = this;
+                this.results.forEach(function (result) {
+                    const equipped = result.equipped;
+                    const dequipped = result.dequipped;
+                    let playerString;
+                    if (self.isCurrentPlayer()) {
+                        playerString = "You";
+                    } else {
+                        playerString = self.entityRef.name;
```

Comment:
I'm not sure we need to have messages for the non-active player equipping/dequipping items. Let's chat about it